--- a/src/env.zig
+++ b/src/env.zig
@@ -22,6 +22,7 @@ pub const isDebug = std.builtin.Mode.Debug == @import("builtin").mode;
 pub const isRelease = std.builtin.Mode.Debug != @import("builtin").mode and !isTest;
 pub const isTest = @import("builtin").is_test;
 pub const isLinux = @import("builtin").target.os.tag == .linux;
+pub const isFreeBSD = @import("builtin").target.os.tag == .freebsd;
 pub const isAarch64 = @import("builtin").target.cpu.arch.isAARCH64();
 pub const isX86 = @import("builtin").target.cpu.arch.isX86();
 pub const isX64 = @import("builtin").target.cpu.arch == .x86_64;
@@ -124,7 +125,7 @@ pub const OperatingSystem = enum {
 
 pub const os: OperatingSystem = if (isMac)
     .mac
-else if (isLinux)
+else if (isLinux or isFreeBSD)
     .linux
 else if (isWindows)
     .windows
